lib/google/gax/paged_enumerable.rb

```diff
@@ -136,17 +138,56 @@ def response
 
       private
 
-      # This inspects a response object and pulls out the first repeated field.
-      def determine_resource_field(response)
-        # Find the first repeated FieldDescriptor on the response Descriptor
-        # Find uses the order of the FieldDescriptor on the Descriptor
-        field = response.class.descriptor.find { |f| f.label == :repeated }
+      def verify_request!
+        page_token = @request.class.descriptor.find do |f|
+          # Google::Protobuf::FieldDescriptorProto::Type::TYPE_STRING = 9
+          f.name == 'page_token' && f.type == 9
+        end
+        if page_token.nil?
+          raise ArgumentError.new(
+            'request must have a page_token field (String)'
+          )
+        end
+
+        page_size = @request.class.descriptor.find do |f|
+          # Google::Protobuf::FieldDescriptorProto::Type::TYPE_INT32 = 5
+          f.name == 'page_size' && f.type == 5
```

Comment:
We should probably allow all the integer types here and not just `TYPE_INT32`.

Reply:
The design doc says:

> Pagination can be inferred when the input message for an RPC contains an `int32` `page_size` field and a `string` `page_token` field.

Reply:
True, but I can assure you that it's over-constrained and I don't see any harm in relaxing it. It has slipped through at least a couple of times: https://github.com/googleapis/googleapis/blob/738b3fa5915ea6000fe5a3054e12bbf8db681662/google/cloud/support/v1alpha1/cloud_support.proto#L163

Reply:
So allow any of the integer types then?

```
TYPE_INT64 = 3
TYPE_UINT64 = 4
TYPE_INT32 = 5
TYPE_FIXED64 = 6
TYPE_FIXED32 = 7
TYPE_UINT32 = 13
TYPE_SFIXED32 = 15
TYPE_SFIXED64 = 16
TYPE_SINT32 = 17
TYPE_SINT64 = 18
```

Reply:
Or limit it to just `TYPE_INT64` and `TYPE_INT32` (3, 5)?

Reply:
@jbolinger I have added int64. Do you want the additional 8 types added as well?

Reply:
The others are much less common. I think these two are fine for now.